src/store/styleguide/pattern.ts

```diff
@@ -147,10 +148,49 @@ export class Pattern {
 	/**
 	 * Returns a property this pattern supports, by its ID.
 	 * @param id The ID of the property.
+	 * @param path If the property you are trying to find is buried inside an object property use the path paremeter to find it.
+	 * eg: `getProperty('image', 'src.srcSet')`.
 	 * @return The property for the given ID, if it exists.
 	 */
-	public getProperty(id: string): Property | undefined {
-		return this.properties.get(id);
+	public getProperty(id: string, path?: string): Property | undefined {
+		if (!path) {
+			return this.properties.get(id);
+		}
+
+		const objectProperty = this.properties.get(id);
+
+		if (!objectProperty) {
+			return;
+		}
+
+		return this.followPath(objectProperty, path, (element, name) => {
+			if (element.getType() !== 'object') {
+				return;
+			}
+
+			return (element as ObjectProperty).getProperty(name);
+		});
+	}
+
+	protected followPath<T>(
```

Comment:
Is this method necessary? Why not just
```
path.split('.').forEach(part => {
	if (property) {
		property = (property as ObjectProperty).getProperty(part);
	}
});
```
without the extra method "followPath" (which also is unnecessarily generic).

Another point: Currently the property nesting relies on the current property value (check for element type 'object'). Instead, the property should yield that it is an object property, like "property.getType() == 'object'", so the final solution could be:
```
path.split('.').forEach(part => {
	if (property) {
		if (property.getType() !== 'object') {
			return undefined;
		}
		property = (property as ObjectProperty).getProperty(part);
	}
});
```

Reply:
done